desktop/libs/azure/src/azure/abfs/upload.py

```diff
@@ -256,49 +258,107 @@ def _get_scheme(self):
       return None
 
 
-class ABFSNewFileUploadHandler(ABFSFileUploadHandler):
+class ABFSNewFileUploadHandler(FileUploadHandler):
   """
-  This handler uploads the file to ABFS if the destination path starts with "ABFS" (case insensitive).
+  This handler uploads the file to Azure Blob File System if the destination path starts with "ABFS" (case insensitive).
   Streams data chunks directly to ABFS.
   """
 
-  def __init__(self, dest_path, username):
+  def __init__(self, fs, dest_path, overwrite):
     self.chunk_size = DEFAULT_WRITE_SIZE
-    self.target_path = None
-    self.file = None
-    self._part_size = DEFAULT_WRITE_SIZE
+    self._fs = fs
+    self.dest_path = dest_path
+    self.overwrite = overwrite
+    self.total_bytes_received = 0
 
-    self.destination = dest_path
-    self.username = username
+  def new_file(self, field_name, file_name, *args, **kwargs):
+    super(ABFSNewFileUploadHandler, self).new_file(field_name, file_name, *args, **kwargs)
 
-    # TODO: _is_abfs_upload really required?
-    if self._is_abfs_upload():
-      self._fs = self._get_abfs(self.username)
-      self.filesystem, self.directory = parse_uri(self.destination)[:2]
+    # Check file extension restrictions
+    is_allowed, err_message = is_file_upload_allowed(file_name)
+    if not is_allowed:
+      raise PopupException(err_message, error_code=400)
 
-    LOG.debug("Chunk size = %d" % DEFAULT_WRITE_SIZE)
+    # Check if the destination path already exists or not
+    if not self._fs.exists(self.dest_path):
+      raise PopupException(f"The destination path {self.dest_path} does not exist.", error_code=404)
 
-  def new_file(self, field_name, file_name, *args, **kwargs):
-    if self._is_abfs_upload():
-      super(ABFSFileUploadHandler, self).new_file(field_name, file_name, *args, **kwargs)
+    # Check if the destination path is a directory or not
+    if not self._fs.isdir(self.dest_path):
+      raise PopupException(f"The destination path {self.dest_path} is not a directory.", error_code=400)
 
-      LOG.info('Using ABFSFileUploadHandler to handle file upload wit temp file%s.' % file_name)
-      self.target_path = self._fs.join(self.destination, file_name)
+    # Check if the file name contains a path separator
+    # This prevents directory traversal attacks
+    if os.path.sep in file_name:
+      raise PopupException("Invalid filename. Path separators are not allowed.", error_code=400)
+
+    # Check if the user has write access to the destination path
+    if not self._fs.check_access(self.dest_path, permission="WRITE"):
+      raise PopupException(f"Insufficient permissions to write to ABFS path {self.dest_path}.", error_code=403)
+
```

Comment:
same here from code lines 277-298 to a new separate fn?

Reply:
Nice suggestion, fixed it